### PR TITLE
Add support for dimension range in createCollection

### PR DESCRIPTION
**Pull request by Hazel-Datastax**
**What this PR does**:
Add support for dimension range in createCollection

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java [hunk 1]
```diff
@@ -380,18 +412,68 @@ private Integer validateModelAndDimension(
                         "Model name '%s' for provider '%s' is not supported",
                         userConfig.modelName(), userConfig.provider()));
 
-    // TODO: is dimension required? do we still auto populate the dimension?
+    // Handle models with a fixed vector dimension
     if (model.vectorDimension().isPresent()) {
       Integer configVectorDimension = model.vectorDimension().get();
       if (userVectorDimension == null) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java:418]: Should we also allow passing `0` to mark "not specified"? (not sure, just asking)

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java [hunk 2]
```diff
@@ -380,18 +412,68 @@ private Integer validateModelAndDimension(
...
+    EmbeddingProvidersConfig.EmbeddingProviderConfig.ValidationType validationType = entry.getKey();
+    List<Integer> validationValues = entry.getValue();
+
+    // Perform validation based on the validation type
+    switch (validationType) {
+      case NUMERIC_RANGE -> {
+        if (userVectorDimension < validationValues.get(0)
+            || userVectorDimension > validationValues.get(1)) {
+          throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+              "The provided dimension value '%s' is not within the supported numeric range",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java:465]: Should also include range: something like

```
"The provided dimension value (%d) is not within the supported numeric range [%d, %d]",
userVectorDimension, validationValues.get(0), validationValues.get(1)
```

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java [hunk 3]
```diff
@@ -380,18 +412,68 @@ private Integer validateModelAndDimension(
...
+              userVectorDimension);
+        }
+      }
+      case OPTIONS -> {
+        if (!validationValues.contains(userVectorDimension)) {
+          throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+              "The provided dimension value '%s' is not within the supported options",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java:472]: Same here, include List of accepted values (can be added via `%s` format parameter)

**Review by tatu-at-datastax — APPROVED** [review]
Looks good, would just like 2 small improvements to error reporting.

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java (hunk 1 above)
Comment (Hazel-Datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java:418]: Hmmm, at least for now, if it's not a fixed dimension, `vectorDimension()` will be null and will not be allowed to pass `0`

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java:418]: ok.

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**